volatility3/framework/layers/qemu.py

```diff
@@ -77,16 +113,19 @@ def _get_ram_segments(self, index: int, page_size: int) -> Tuple[List[Tuple[int,
         base_layer = self.context.layers[self._base_layer]
 
         while not done:
-            addr = self.context.object(self._qemu_table_name + constants.BANG + 'unsigned long long',
-                                       offset = index,
-                                       layer_name = self._base_layer)
+            # Use struct.unpack here for performance improvements
+            addr = struct.unpack('>Q', base_layer.read(index, 8))[0]
+
             # Flags are stored in the n least significant bits, where n equals the bit-length of pagesize
             flags = addr & (page_size - 1)
             # addr equals the highest multiple of pagesize <= offset
             # (We assume that page_size is a power of 2)
             addr = addr ^ (addr & (page_size - 1))
             index += 8
 
+            if addr > self._pci_hole_start:
```

Comment:
Should be `>=` I think? Since `addr` marks the start of a new block, which would otherwise run into the gap?

Reply:
Cool, I'll fix that.